### PR TITLE
adding resubmit flag to scan create

### PR DESCRIPTION
**Pull request by tiagobcx**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

- Adding resubmit flag under scan create command

### References

- https://checkmarx.atlassian.net/browse/AST-14554

### Testing

- Unit and integration tests

### Checklist

- [x] I have added documentation for new/changed functionality in this PR (if applicable).
- [x] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used

### CODE REVIEW

File: internal/commands/scan.go [hunk 1]
```diff
@@ -597,30 +596,76 @@ func setupScanTypeProjectAndConfig(
 	info["project"].(map[string]interface{})["id"] = projectID
 	// Handle the scan configuration
 	var configArr []interface{}
-	if _, ok := info["config"]; !ok {
-		err = json.Unmarshal([]byte("[]"), &configArr)
-		if err != nil {
-			return err
+	resubmit, _ := cmd.Flags().GetBool("resubmit")
```
Comment (diogopcx) [inline internal/commands/scan.go:599]: Let's use the commonParams.ScanResubmit constant here.

File: internal/commands/scan.go [hunk 2]
```diff
@@ -597,30 +596,76 @@ func setupScanTypeProjectAndConfig(
...
+	if resubmit {
+		logger.PrintIfVerbose("using latest scan configuration due to --resubmit flag")
```
Comment (diogopcx) [inline internal/commands/scan.go:601]: Same here with PrintfIfVerbose(logger.PrintIfVerbose("using latest scan configuration due to --%s flag", commonParams.ScanResubmit)

File: internal/commands/scan.go [hunk 3]
```diff
@@ -597,30 +596,76 @@ func setupScanTypeProjectAndConfig(
 	info["project"].(map[string]interface{})["id"] = projectID
 	// Handle the scan configuration
 	var configArr []interface{}
-	if _, ok := info["config"]; !ok {
-		err = json.Unmarshal([]byte("[]"), &configArr)
-		if err != nil {
-			return err
+	resubmit, _ := cmd.Flags().GetBool("resubmit")
+	if resubmit {
```
Comment (diogopcx) [inline internal/commands/scan.go:600]: We can simplify the if/else by extracting the common part for each engine in the end of each block to outside the if/else

File: internal/commands/scan.go [hunk 4]
```diff
@@ -1697,6 +1789,8 @@ func toScanView(scan *wrappers.ScanResponseModel) *scanView {
 	var origin string
 	var scanType string
 	var scanTimeOut string
+	fmt.Printf("%v", scan.Engines)
+	fmt.Printf("configs:%v", scan.Metadata.Configs)
```
Comment (diogopcx) [inline internal/commands/scan.go:1793]: Remove these left over prints

File: test/integration/scan_test.go [hunk 5]
```diff
@@ -32,6 +32,8 @@ const (
 	scanCommand           = "scan"
 	kicsRealtimeCommand   = "kics-realtime"
 	invalidEngineValue    = "invalidEngine"
+	scanList              = "list"
+	projectIDParams = "project-id="
```
Comment (diogopcx) [inline test/integration/scan_test.go:36]: Format this line

File: test/integration/scan_test.go [hunk 6]
```diff
@@ -431,7 +433,7 @@ func listScanByID(t *testing.T, scanID string) []wrappers.ScanResponseModel {
 	outputBuffer := executeCmdNilAssertion(
 		t,
 		"Getting the scan should pass",
-		"scan", "list", flag(params.FormatFlag), printer.FormatJSON, flag(params.FilterFlag), scanFilter,
+		"scan",scanList, flag(params.FormatFlag), printer.FormatJSON, flag(params.FilterFlag), scanFilter,
```
Comment (diogopcx) [inline test/integration/scan_test.go:436]: Format this line

Comment (tiagobcx) [reply test/integration/scan_test.go]: Done, thanks !

File: test/integration/scan_test.go (hunk 5 above)
Comment (tiagobcx) [reply test/integration/scan_test.go]: Done, thanks !

File: internal/commands/scan.go (hunk 4 above)
Comment (tiagobcx) [reply internal/commands/scan.go]: Done, thanks !

File: internal/commands/scan.go (hunk 3 above)
Comment (tiagobcx) [reply internal/commands/scan.go:601]: Done, thanks !

File: internal/commands/scan.go (hunk 2 above)
Comment (tiagobcx) [reply internal/commands/scan.go]: Done, thanks !

File: internal/commands/scan.go (hunk 1 above)
Comment (tiagobcx) [reply internal/commands/scan.go]: Done, thanks !

**Review by diogopcx — APPROVED** [review]

**Pull request merged**